src/cargo/core/compiler/custom_build.rs

```diff
@@ -413,8 +413,10 @@ fn build_work(cx: &mut Context<'_, '_>, unit: &Unit) -> CargoResult<Job> {
     let nightly_features_allowed = cx.bcx.config.nightly_features_allowed;
     let extra_check_cfg = cx.bcx.config.cli_unstable().check_cfg;
     let targets: Vec<Target> = unit.pkg.targets().to_vec();
+    let msrv = unit.pkg.rust_version().map(|v| v.clone());
     // Need a separate copy for the fresh closure.
     let targets_fresh = targets.clone();
+    let msrv_fresh = msrv.clone();
```

Comment:
why `msrv` and `msrv_fresh`?

Reply:
The unit was moved into the first closure, so it needs to be cloned. Therefore, Option<&T> cannot be passed.

Reply:
Think you might have replied to the wrong comment

Reply:
```console
❯ cargo b
   Compiling cargo v0.77.0 (/Volumes/t7/code/cargo)
error[E0521]: borrowed data escapes outside of function
   --> src/cargo/core/compiler/custom_build.rs:435:17
    |
251 |   fn build_work(cx: &mut Context<'_, '_>, unit: &Unit) -> CargoResult<Job> {
    |                                           ----  - let's call the lifetime of this reference `'1`
    |                                           |
    |                                           `unit` is a reference that is only valid in the function body
...
435 |       let dirty = Work::new(move |state| {
    |  _________________^
436 | |         // Make sure that OUT_DIR exists.
437 | |         //
438 | |         // If we have an old build directory, then just move it into place,
...   |
575 | |         Ok(())
576 | |     });
    | |      ^
    | |      |
    | |______`unit` escapes the function body here
    |        argument requires that `'1` must outlive `'static`

```

Reply:
Thanks for your review!  💚 💙 💜 💛 ❤️

Reply:
> Think you might have replied to the wrong comment
Oh yes. The reason for msrv_fresh is we also moved it into the first closure.